### PR TITLE
DM-47612: Refactoring and simplifications in the Qserv worker replication service

### PR DESCRIPTION
**Pull request by iagaponenko**

### CODE REVIEW

File: src/replica/worker/WorkerReplicationRequest.h [hunk 1]
```diff
@@ -102,86 +98,13 @@ class WorkerReplicationRequest : public WorkerRequest {
      */
     void setInfo(ProtocolResponseReplicate& response) const;
 
-    bool execute() override;
+    virtual bool execute();
```
Comment (jgates108) [inline src/replica/worker/WorkerReplicationRequest.h:101]: Shouldn't it be `override`?

File: src/replica/worker/WorkerSqlRequest.h [hunk 2]
```diff
@@ -92,7 +92,7 @@ class WorkerSqlRequest : public WorkerRequest {
      */
     void setInfo(ProtocolResponseSql& response) const;
 
-    bool execute() override;
+    virtual bool execute();
```
Comment (jgates108) [inline src/replica/worker/WorkerSqlRequest.h:95]: Shouldn't it be `override`?

**Review by jgates108 — APPROVED** [review]
Looks good, just minor comments. It does simplify the code significantly :)

File: src/replica/worker/WorkerDeleteRequest.h [hunk 3]
```diff
@@ -86,55 +80,20 @@ class WorkerDeleteRequest : public WorkerRequest {
      */
     void setInfo(ProtocolResponseDelete& response) const;
 
-    bool execute() override;
+    virtual bool execute() override;
```
[the remark this answers is not in the document]
Comment (iagaponenko) [reply src/replica/worker/WorkerDeleteRequest.h:83]: Good point.

**Pull request merged**